### PR TITLE
Add user-to-user matching algorithm

### LINKED ISSUE #59: Add user-to-user matching algorithm
The user-to-user matching algorithm will give each user a score against another user based on their common hobbies.

### PR DESCRIPTION
**Pull request by nadavsu**
# Description
Added a function which returns a matching score between two users which is used to rank the different apartments for the seeker. It works by dividing the number of similar hobbies shared between the two users by the total number of hobbies between the two users. Returns a number between 0 and 1.

## Manual Tests
All users with the same score - passed
4 users with different scores with a QuerySet limit of 3 apartments - passed
4 users with different scores and a QuerySet limit of 3 apartments - passed
3 users with different scores and a QuerySet size limit of 3 apartments - passed

## Additional Information
`get_all_relevant_apartments` is a temporary function which was written in order to create this feature. Once the search feature is created, this function must be deleted, as we'll use the function in the search feature.

### Issues closed by this PR
Fixes #59

### CODE REVIEW

**Review by DanielMy777 — APPROVED** [review]
LGTM 
Are we waiting until the search is merged before we merge this?

**Review by Micha546 — APPROVED** [review]
LGTM

**Comment by nadavsu** [comment]
> LGTM
> Are we waiting until the search is merged before we merge this?

Nope. We need to add some tests to this.
But before that we need to add tests for the hobbies
and before that we need to add hobbies to the signup form.

**Review by AmitAharoni — APPROVED** [review]

**Pull request merged**
